### PR TITLE
Enh ext dlwp healpix

### PR DESCRIPTION
**Pull request by daviddpruitt**
# Modulus Pull Request

## Description
Adds the dlwp healpix model from the paper Advancing Parsimonious Deep Learning Weather Prediction using the HEALPix Mesh
(https://essopenarchive.org/doi/full/10.22541/essoar.169603505.58030377)

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/modulus/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/modulus/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/modulus/issues) is linked to this pull request.

## Dependencies
No additional dependencies

### CODE REVIEW

File: examples/weather/dlwp_healpix/trainer.py [hunk 1]
```diff
@@ -0,0 +1,480 @@
...
+                self.writer.add_scalar(tag="val_loss", scalar_value=validation_error, global_step=iteration)
+                
+                # Per-variable loss
+                for v_idx, v_name in enumerate(self.output_variables):
+                    self.writer.add_scalar(tag=f"val_loss/{v_name}", scalar_value=validation_errors[v_idx],
+                                           global_step=iteration)
+
+                # Write model checkpoint to file, using a separate thread
+                self.logger0.info("Writing checkpoint")
+                thread = threading.Thread(
```
Comment (akshaysubr) [inline examples/weather/dlwp_healpix/trainer.py:446]: Should switch to the modulus checkpointing format as discussed in the meeting last week: https://github.com/NVIDIA/modulus/blob/main/modulus/launch/utils/checkpoint.py#L174

**Review by akshaysubr — APPROVED** [review]
@daviddpruitt This PR looks mostly good. Only main concern I have is that the checkpointing in the example is a custom format and not using the modulus checkpointing format. The latter is needed to be able to enable easy inference and we should switch to that in this PR. I think everything else can potentially be in follow up PRs.

File: modulus/models/dlwp_healpix/HEALPixRecUNet.py [hunk 2]
```diff
@@ -0,0 +1,471 @@
...
+    func_torch: bool = False
+    auto_grad: bool = False
+
+
+class HEALPixRecUNet(Module):
+    """Deep Learning Weather Prediction (DLWP) recurrent UNet model on the HEALPix mesh."""
+
+    def __init__(
+        self,
+        encoder: DictConfig,
```
Comment (loliverhennigh) [inline modulus/models/dlwp_healpix/HEALPixRecUNet.py:54]: One potential issue is that this encoder config will not be json serializable. Not sure though. We might need to convert it to a normal python dict if this is the case.

**Review by loliverhennigh — APPROVED** [review]

File: examples/weather/dlwp_healpix/trainer.py (hunk 1 above)
Comment (daviddpruitt) [reply examples/weather/dlwp_healpix/trainer.py:446]: Updated to modulus checkpointing

File: examples/weather/dlwp_healpix/trainer.py [hunk 3]
```diff
@@ -0,0 +1,480 @@
...
+                self.gscaler.scale(self.static_loss_train).backward()
+            
+            # sync here
+            capture_stream.synchronize()
+
+            gc.collect()
+            torch.cuda.empty_cache()
+
+            # create graph
+            self.train_graph = torch.cuda.CUDAGraph()
```
[the remark this answers is not in the document]
Comment (daviddpruitt) [reply examples/weather/dlwp_healpix/trainer.py:191]: Will do, once this goes in I'll update with `StaticCaptureTraining`

**Pull request merged**